src/robot/Robot.cpp

```diff
@@ -1,21 +1,454 @@
+  auto subRobot = std::make_shared<Robot>(metaSkeleton, name);
+  const auto& root = mRootRobot ? mRootRobot : RobotPtr(this);
+  subRobot->setRootRobot(root);
+  subRobot->setWorld(nullptr); // inherit root robot world
+  mSubRobots[name] = subRobot;
+  return subRobot;
+}
+
+//=============================================================================
+Eigen::VectorXd Robot::getNamedConfiguration(const std::string& name) const
+{
+  auto configuration = mNamedConfigurations.find(name);
+  if (configuration == mNamedConfigurations.end())
+    return Eigen::VectorXd();
+
+  return configuration->second;
+}
+
+//=============================================================================
+void Robot::setNamedConfigurations(
+    std::unordered_map<std::string, const Eigen::VectorXd> namedConfigurations)
+{
+  mNamedConfigurations = std::move(namedConfigurations);
+}
+
+//=============================================================================
+trajectory::TrajectoryPtr Robot::planToConfiguration(
+    const Eigen::VectorXd& goalConf,
+    const constraint::TestablePtr& testableConstraint,
+    const std::shared_ptr<planner::ConfigurationToConfigurationPlanner>&
+        planner,
+    const std::shared_ptr<aikido::planner::TrajectoryPostProcessor>
+        trajPostProcessor) const
+{
+  statespace::dart::MetaSkeletonStateSpace::State* goalState = nullptr;
+  try
+  {
+    mStateSpace->convertPositionsToState(goalConf, goalState);
+  }
+  catch (const std::exception& e)
+  {
+    dtwarn << "Cannot convert configuration to robot state: " << e.what()
+           << std::endl;
+    return nullptr;
+  }
+
+  // Create the problem.
+  auto problem = planner::dart::ConfigurationToConfiguration(
+      mStateSpace,
+      mStateSpace->getScopedStateFromMetaSkeleton(mMetaSkeleton.get()),
+      goalState,
+      testableConstraint);
+
+  // Default to base Snap Planner
+  auto basePlanner = planner;
+  if (!basePlanner)
+  {
+    using planner::SnapConfigurationToConfigurationPlanner;
+    using statespace::GeodesicInterpolator;
+    basePlanner = std::make_shared<SnapConfigurationToConfigurationPlanner>(
+        mStateSpace, std::make_shared<GeodesicInterpolator>(mStateSpace));
+  }
+
+  // Convert to DART Planner
+  using planner::dart::
+      ConfigurationToConfiguration_to_ConfigurationToConfiguration;
+  auto dartPlanner = std::make_shared<
+      ConfigurationToConfiguration_to_ConfigurationToConfiguration>(
+      basePlanner, mMetaSkeleton);
+
+  // Solve the problem with the DART planner.
+  auto rawPlan = dartPlanner->plan(problem, /* result */ nullptr);
+
+  // Postprocess if enabled or provided
+  auto postprocessor
+      = (trajPostProcessor)
+            ? trajPostProcessor
+            : ((mEnablePostProcessing) ? mDefaultPostProcessor : nullptr);
+  if (rawPlan && postprocessor)
+  {
+    // Cast to interpolated or spline:
+    auto interpolated
+        = dynamic_cast<const aikido::trajectory::Interpolated*>(rawPlan.get());
```

Comment:
It seems like rawPlan.get() will either return Interpolated or Spline, and you're not sure which, so you try both? I'm wondering if there might be a cleaner way to do this (i.e., where you store what type rawPlan.get() will return)? No need to do it if it is too much work, I just wanted to mention that this current system of trying two casts to see which succeeds seems a bit strange imo